src/api/layers/ExternalLayer.class.js

```diff
@@ -58,10 +78,18 @@ export default class ExternalLayer extends AbstractLayer {
         this.baseURL = baseURL
         this.abstract = abstract
         this.extent = extent
+        this.legends = legends
         this.isLoading = isLoading
     }
 
     getURL() {
         return this.baseURL
     }
+
+    get hasLegend() {
+        if (this.abstract || this.legends.length > 0) {
+            return true
+        }
+        return super.hasLegend
```

Comment:
Might be simplified to
`return this.abstract || this.legends.length > 0 || super.hasLegend`
so that there's no `true` being returned